File: lib/limelight/players/text_area.rb
#- Copyright 2008 8th Light, Inc.
#- Limelight and all included source files are distributed under terms of the GNU LGPL.

module Limelight
  module Players
    
    module TextArea
      class << self

        def extended(prop)
          text_area = Limelight::UI::Model::Inputs::TextAreaPanel.new
          prop.panel.add(text_area)
          prop.text_area = text_area.text_area
          set_default_styles(prop)
        end
        
        def set_default_styles(prop)
          prop.style.width ||= "200"
          prop.style.height ||= "88"
        end

      end

      attr_accessor :text_area
  
    end
  
  end
end

File: src/limelight/ui/model/inputs/TextArea.java
//- Copyright 2008 8th Light, Inc. All Rights Reserved.
//- Limelight and all included source files are distributed under terms of the GNU LGPL.

package limelight.ui.model.inputs;

import javax.swing.*;
import java.awt.*;

public class TextArea extends JTextArea
{
  private TextAreaPanel panel;

  public TextArea(TextAreaPanel panel)
  {
    this.panel = panel;
    setLineWrap(true);
    setBorder(BorderFactory.createEtchedBorder(Color.lightGray, Color.gray));
  }

  public void repaint()
  {
    if(panel != null && panel.getRoot() != null)
      panel.getRoot().addDirtyRegion(panel.getAbsoluteBounds());
  }

  public void repaint(long tm, int x, int y, int width, int height)
  {
    repaint(new Rectangle(x, y, width, height));
  }

  public void repaint(Rectangle dirtyBounds)
  {
    if(panel != null && panel.getRoot() != null)
    {
      Point location = panel.getAbsoluteLocation();
      dirtyBounds.translate(location.x, location.y);
      panel.getRoot().addDirtyRegion(dirtyBounds);
    }
  }

  public boolean isShowing()
  {
    return true;
  }
}
